### PR TITLE
Add ec point validation functions

### PR DESCRIPTION
**Pull request by nomick**
Adds two more functions to validate EC points received from untrusted locations.

### CODE REVIEW

File: openssl/src/ec.rs [hunk 1]
```diff
@@ -527,6 +527,30 @@ impl EcPointRef {
             .map(|_| ())
         }
     }
+
+    /// Checks if point is infinity
```
Comment (sfackler) [inline openssl/src/ec.rs:531]: The docs should mention what these correspond to like on the other methods in the module.

File: openssl/src/ec.rs [hunk 2]
```diff
@@ -527,6 +527,30 @@ impl EcPointRef {
...
+    pub fn is_infinity(&self, group: &EcGroupRef) -> Result<bool, ErrorStack> {
```
Comment (sfackler) [inline openssl/src/ec.rs:532]: This is not documented to ever return an error, just 0 or 1.

File: openssl/src/ec.rs (hunk 1 above)
Comment (nomick) [reply openssl/src/ec.rs:531]: added

File: openssl/src/ec.rs (hunk 2 above)
Comment (nomick) [reply openssl/src/ec.rs]: Just wanted to make sure, but I see that it does not correspond to other functions..

**Pull request merged**